src/parser/parser.yy

```diff
@@ -1638,14 +1643,46 @@ drop_edge_sentence
     }
     ;
 
+index_field
+    : name_label {
+        $$ = new meta::cpp2::IndexFieldDef();
+        $$->set_name($1->c_str());
+        delete $1;
+    }
+    | name_label L_PAREN INTEGER R_PAREN {
+        if ($3 > std::numeric_limits<int16_t>::max()) {
+            throw nebula::GraphParser::syntax_error(@3, "Out of range:");
+        }
+        $$ = new meta::cpp2::IndexFieldDef();
+        $$->set_name($1->c_str());
+        $$->set_type_length($3);
+        delete $1;
+    }
+    ;
+
+index_field_list
+    : index_field {
+        $$ = new IndexFieldList();
+        std::unique_ptr<meta::cpp2::IndexFieldDef> field;
+        field.reset($1);
+        $$->addField(std::move(field));
```

Comment:
I think you could accept raw pointer as parameter directly.

Reply:
> I think you could accept raw pointer as parameter directly.

I think here is ok. we don't need to keep this pointer. unique_ptr is enough.